[spaceship] Improve README

janpio opened the pull request:
<!-- Thanks for contributing to _fastlane_! Before you submit your pull request, please make sure to check the following boxes by putting an x in the [ ] (don't: [x ], [ x], do: [x]) -->

### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context
spaceship documentation makes it sound as this was a super new thing. this is not the case any more, as fastlane has been using spaceship for quite some time.

### Description
This PR rewrites the text of the spaceship documentation to reflect that spaceship is not a super new thing any more (and removes the focus on "vs. scraping" a bit)

janpio committed:
remove links to other tools

fastlane-bot-helper commented:
<!--
  0 Errors
  1 Warning: Pull Request is Work in Progre...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Pull Request is Work in Progress</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

janpio committed:
Update README.md

Integrate information from https://spaceship.airforce/ and reword so it becomes clearer that spaceship is not a super new thing any more

janpio committed:
docs CI

janpio committed:
"Support for CI machines" logic and grammar

joshdholtz reviewed (approved):
🔥
